Added styles for uimessage control

dpavloff opened the pull request:
Added styles for flexberry-uimessage components: new icon, removed border, slightly changed font props

dpavloff committed:
Added necessary files for uimessage, removed borders


Added styles for uimessage


Added importants


Added uimessage-associate, sorted controls alphabetically

dpavloff committed:
Removed spinners from number input in flexberry-field (FF)

dpavloff commented:
23.07.2020
Also removed arrow spinners from flexberry-field [type="number"] (firefox)

dpavloff committed:
Added flex-uimessage: 'ghost' to theme.config

KollegOFF commented on src/themes/ghost/components/flexberry-field.overrides:
```diff
@@ -8,6 +8,10 @@
     border-color: @fieldReadonlyBorderColor;
 }
 
+.ui.form input[type="number"] {
+    -moz-appearance: textfield;
+}
```
Давайте без этого?

dpavloff replied:
Всё таки не стоит убирать стрелочки на number input'ах?

KollegOFF replied:
Я думаю что не стоит, при желании, это всегда можно сделать в приложении.

pepelyaeva replied:
https://github.com/Flexberry/ember-flexberry-themes/blob/develop/src/definitions/components/flexberry-objectlistview.less#L100 вот тут тоже тогда убрать? или убрать стрелки только для олва? они будут мешать в поле для перехода на страницу.

KollegOFF replied:
Лучше убрать только для олва в теме `ghost`, в `blue-sky` кнопка отдельно, не будут мешать.

dpavloff replied:
В таком случае, я сделаю более конкретный селектор, чтобы попадал только на пейджинг олва

pepelyaeva replied:
Удали тогда, пожалуйста, еще [здесь](https://github.com/Flexberry/ember-flexberry-themes/blob/develop/src/definitions/components/flexberry-objectlistview.less#L100) и добавь -webkit-appearance

dpavloff replied:
Исправил.

dpavloff committed:
More precise selectors for OLV-input

- changed location for webkit-appearance: none;
- changed selector for one with more precision

KollegOFF commented on src/themes/ghost/components/flexberry-field.overrides:
```diff
@@ -8,6 +8,15 @@
     border-color: @fieldReadonlyBorderColor;
 }
 
+.object-list-view-container~.ui.secondary.menu .ui.input.nav-bar-search-page input[type="number"] {
+    -moz-appearance: textfield;
+}
+
+.ui.input.nav-bar-search-page input[type="number"]::-webkit-outer-spin-button,
+.ui.input.nav-bar-search-page input[type="number"]::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+}
```
Что-то я не понял, тут для разных браузеров разные селекторы?
И у нас вроде `autoprefixer` где-то был...

pepelyaeva replied:
Да, для них по-разному нужно https://stackoverflow.com/questions/23372903/hide-spinner-in-input-number-firefox-29. Можно сделать как-то так

```
.ui.input.nav-bar-search-page input[type='number'] {
  -moz-appearance: textfield;/*For FireFox*/

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button { /*For Webkits like Chrome and Safari*/
    -webkit-appearance: none;
    margin: 0;
  }
}
```

dpavloff replied:
Йеп. В FF это почему-то правило, а в хроме и сафари - псевдокласс. Меня больше смущает то, что селектор от первого не подходит ко второму... Но я думаю это можно исправить.

pepelyaeva commented on src/themes/ghost/components/flexberry-uimessage.overrides:
```diff
@@ -0,0 +1,19 @@
+/* Primary */
+.ui.message {
+    box-shadow: @uimessageBoxShadow;
+}
```
Нужно поиспользовать семантиковые переменные @errorBoxShadow, @warningBoxShadow и тд, и у брать эти стили.

dpavloff replied:
Заменил значение переменной с "0 0 0 0 transparent !important" на @errorBoxShadow без !important. Вместо этого добавил 19 селекторов на каждый возможный вариант .ui.message - считаю, что это хороший обмен.

pepelyaeva replied:
Зачем селекторы? Можно просто переменные переопределить же и стилей не писать

dpavloff replied:
Моей целью было избавиться от !important - такое решение, вроде как, работает. Или я не до конца понял?

pepelyaeva commented on src/themes/ghost/components/flexberry-uimessage.overrides:
```diff
@@ -0,0 +1,19 @@
+.ui.message .header {
+    font-family: @uimessageFont;
+    font-weight: @uimessageFontWeight;
+}
```
Здесь тоже можно обойтись переменными @headerFont, @headerFontWeight

pepelyaeva commented on src/definitions/components/flexberry-objectlistview.less:
```diff
@@ -97,7 +97,6 @@
 
 input::-webkit-outer-spin-button,
 input::-webkit-inner-spin-button {
-  -webkit-appearance: none;
   margin: 0;
 }
```
Вот это давай тоже перенесем к -webkit-appearance: none;

dpavloff replied:
Перенёс.

dpavloff committed:
PR Fix


Fixed border for ui-message


Small refactoring

dpavloff committed:
Added component to blueprints

KollegOFF commented on src/themes/ghost/components/flexberry-field.overrides:
```diff
@@ -8,11 +8,20 @@
     border-color: @fieldReadonlyBorderColor;
 }
 
+.ui.input.nav-bar-search-page input[type='number'] {
+  -moz-appearance: textfield;/*For FireFox*/
+
+  &::-webkit-inner-spin-button,
+  &::-webkit-outer-spin-button { /*For Webkits like Chrome and Safari*/
+    -webkit-appearance: none;
+    margin: 0;
+  }
+}
```
Думаю это больше относится к стилям `flexberry-objectlistview`.
